python-avd/schema_tools/generate_classes/src_generators.py

```diff
@@ -0,0 +1,460 @@
+    """
+
+    name: str
+    type_hint: FieldTypeHintSrc
+    value: str
+    description: str | None = None
+    """The default assignment source code for assigning lists or dicts as default values."""
+
+    def __str__(self) -> str:
+        """
+        Render source code for one field.
+
+        <name>: ClassVar[<type-hint>] = <value>
+        """
+        return f"{self.name}: ClassVar[{self.type_hint!s}] = {self.value}"
+
+    def _docstring(self) -> str:
+        """Render the content of the docstring for this field as source code."""
+        if not self.description:
+            return ""
+
+        docstring = "\n".join(wrap(self.description, width=100, replace_whitespace=False))
+        return f"{docstring}\n"
+
+    def get_imports(self) -> set:
+        imports = {"from typing import ClassVar"}
+        imports.update(self.type_hint.get_imports())
+        return imports
+
+
+@dataclass
+class ModelSrc:
+    """
+    Dataclass containing the elements to generate Python source code for this class including any nested classes and fields.
+
+    Can have nested classes and fields.
+
+    Use str() on the instance to render the source code.
+    """
+
+    name: str
+    classes: list[ModelSrc | CollectionSrc]
+    fields: list[FieldSrc]
+    class_vars: list[ClassVarSrc] | None = None
+    imports: set[str] | None = None
+    base_classes: list[str] | None = None
+    description: str | None = None
+    allow_extra: bool = False
+
+    def __str__(self) -> str:
+        """Renders the Python source code for this class including any nested classes and fields."""
+        base_classes = ", ".join(self.base_classes) if self.base_classes else BASE_MODEL_NAME
+
+        classsrc = f"class {self.name}({base_classes}):\n" if base_classes else f"class {self.name}:\n"
+
+        if self.description:
+            description = "\n".join(wrap(self.description, width=100, replace_whitespace=False))
+            if "\n" in description:
+                classsrc += indent(f'"""\n{description.strip()}\n"""\n', "    ")
+            else:
+                classsrc += indent(f'"""{description}"""\n', "    ")
+
+        if classes := self._render_classes():
+            classsrc += f"{classes}\n"
+
+        if class_vars := self._render_class_vars():
+            classsrc += f"{class_vars}\n"
+
+        if fields := self._render_fields():
+            classsrc += f"{fields}\n"
+
+        if not classes and not fields:
+            classsrc += "    pass\n"
+
+        return classsrc
+
+    def _init_docstring(self) -> str:
+        """Render the docstring for the __init__ method for this model as source code."""
+        docstring_elements = [f"{self.name}.\n"]
+        if self.description:
+            # Build docstring styled description
+            docstring_elements.append("\n".join(wrap(self.description, width=100, replace_whitespace=False)))
+
+        if self.fields:
+            args = "Args:\n-----\n"
```

Comment:
```suggestion
            args = "Args:\n-----\n"
```

this is not following a conventional docstring pattern (but maybe we don't care) so it can make doc generation harder if we ever need this

Reply:
Fixed